### DISCUSSION TITLE
[Question] Handling many attributes per daily note

### DISCUSSION THREAD

**Discussion by PAK90** [TriliumNext/Trilium#6844] [Q&A] [answered]
I really like the look of Trilium; I've been looking for personal logging software for a while now. One thing that I want is many daily attributes, like the 'weight' example one. And by 'many' I mean probably 15+. Another thing I want to store is series data, for example an array of hourly temperature values. I could store this by using a string label presumably (unless it has a length limit), but the main problem with 15+ daily labels is the UI would get really crowded. Is there any alternative/preferred solution for this?

**Comment by zadam** [comment, marked as answer]
Hi, currently UI with a lot of promoted attributes can be busy, this was discussed in #1037 

But you mention series data and for that I'm not sure if labels are the best way to store it. Personally I do something very similar to what you do - storing quite a lot of daily data - I store them into per-day "data" note (actually JSON code note). See e.g. https://zadam.github.io/trilium/backend_api/BackendScriptApi.html#createDataNote

Advantage of this approach is that you can also easily store structured data. Disadvantage is that you need to script it which requires some programming effort ... (see https://github.com/zadam/trilium/wiki/Scripts )

**Comment by PAK90** [comment]
Oh that's cool, I didn't realise data notes were an option! That should solve a lot of issues. Is it possible to explicitly link a data note to a daily note? Or would it be implied by the creation date of the data note?

As for #1037 , will it be part of 0.43, and if so, when might we see 0.43? I might try to make a PR for this specific issue if 0.43 will still be a while.

**Comment by zadam** [comment]
> Is it possible to explicitly link a data note to a daily note

I simply create data note under a daily note. Or if that's messy for you, you can also create in some other place and link it through a relation attribute. For some other things I create a label `date=2020-05-27` using which I can retrieve it ...

Since this is something you kind of need to script, it's up to you how exactly you do it ..

> As for [#1037](https://github.com/zadam/trilium/issues/1037) , will it be part of 0.43, and if so, when might we see 0.43?

In a couple of weeks probably.

**Comment by Suncatcher** [comment]
> Personally I do something very similar to what you do - storing quite a lot of daily data - I store them into per-day "data" note (actually JSON code note). See e.g. https://zadam.github.io/trilium/backend_api/BackendScriptApi.html#createDataNote

@zadam are data notes still an option? I didn't find mentioning about them in the [Trilium wiki](https://github.com/TriliumNext/Trilium/wiki) and [Trilium help](https://triliumnext.github.io/Docs/)

**Comment by eliandoran** [comment]
@Suncatcher , they were removed somewhere in 2021. The implementation of the API is really simple, it just created a code note whose content was the JSON representation of the data you were passing to it. 

You can see the implementation here: 79e7762c723215b3f5b69b44dfc1f71409e47729

Nowadays you can also do the same with attachments, which does a better job of hiding the file from the tree, depending on the use case.

**Reply by Suncatcher** [reply]
is there any note type to store multiple attributes except attaching a JSON file? for me creating a JSON file is an extra step I'd like to avoid

**Reply by eliandoran** [reply]
@Suncatcher , no. Attributes are metadata for the note, not for the note content. It doesn't have to be JSON, it can be a simple text file like:

```
attribute1=value1
attribute2=value2
```
